src/issueFinder.ts

```diff
@@ -22,16 +22,20 @@ export class IssueFinder {
         labels: string[],
         repositoryIdentifiers: string[]
     ): Promise<Issue[]> {
-        // TODO: loop through all the labels
-        const label = labels.shift() || '';
-        const result = await this.fetcher.fetchMatchingIssues(token, label, repositoryIdentifiers);
+        let result;
+        for (let label of labels) {
+            result = await this.fetcher.fetchMatchingIssues(token, label, repositoryIdentifiers);
```

Comment:
Regardless of the type of loop, I think you'll need to save each fetcher result somewhere before moving on to the next one.